### ISSUE TITLE
Add OKX Chain and become the official maintainer for this network integration feature

### ISSUE THREAD

**Issue by Jalal-1** [humanprotocol/human-protocol#122]
Human Protocol supports arbitrary EVM networks but there's a bit of work to do to make a network first class and make sure that it works. All of our EVM network launches are payable in ETH and you can gain a yearly bonus by being our network sponsor. If you need support please join tech-discussions on the human protocol discord linked on hmt.ai.

### Steps to receive a bounty
- [ ] Suggest and have us authorize your bridging strategy. After all we need to bridge HMT over to the new network. Usually, we need to use an erc20-compatible approach.
- [ ] Deploy the contracts on testnet
- [ ] Verify the contracts on the testnet block scanner
- [ ] Make sure that everything is working on testnet
- [ ] Add the new deployed contracts addresses to this [list](https://github.com/humanprotocol/human-protocol/blob/main/CONTRACTS_LIST.md)
- [ ] Transfer ownership of testnet contracts
- [ ] Launch a working app on the network. The easiest one to get going is our example [fortune](https://github.com/humanprotocol/human-protocol/tree/main/packages/examples/fortune). Make a video of the functionality of the app working on the new network and link to relevant scanner transactions.
- [ ] Deploy the contracts on mainnet
- [ ] Verify the contracts on the mainnet block scanner
- [ ] Make sure that everything is working on mainnet
- [ ] Add the new deployed contracts addresses to this [list](https://github.com/humanprotocol/human-protocol/blob/main/CONTRACTS_LIST.md)
- [ ] Transfer ownership of mainnet contracts
- [ ] Create a subgraph with this [sdk](https://github.com/humanprotocol/human-protocol/tree/main/packages/sdk/typescript/subgraph)
- [ ] Add to [human dashboard](https://dashboard.humanprotocol.org)
- [ ] Request to become the maintainer for your EVM network. We will list you on our repo, give you an extra reward (generally around 1 ETH per year) and invite you to conferences. We will celebrate your name forever!
- [ ] Bridge HMT from Ethereum
- [ ] Verify bridging of tokens works and link to relevant scanner transactions.
- [ ] Add the chain to typescript [SDK](https://github.com/humanprotocol/human-protocol/blob/develop/packages/sdk/typescript/human-protocol-sdk/src/constants.ts)
- [ ] Add the chain to python [SDK](https://github.com/humanprotocol/human-protocol/blob/develop/packages/sdk/python/human-protocol-sdk/human_protocol_sdk/constants.py)

### High level rules to make sure that you are doing this correctly
- [ ] We never want to create a mintable token. Your change should not change the total token supply from 1 billion
- [ ] When deploying production contracts use a dedicated wallet with a seed phrase you don't use anywhere else. You might be required to share it with us.

**Comment by Vonder85** [comment]
Hello, I am interested to work on it. I'm a friend of @Charlie85270. 
I will use Anyswap as bridge strategy, can i start the work ?

**Comment by foufrix** [comment]
Hi @Jalal-1 did @Vonder85 received an anwser?

@Vonder85 sorry for late reply. Do not hesitate to reach through Discord in `tech-discussion` in case there is no reply on Github.

**Comment by SteffQing** [comment]
Hey guys. I have started working on the integration for OKC and I will be sharing my updates very soon
I will be using MultiChain as my bridging strategy

**Comment by foufrix** [comment]
Bridge is deployed at https://app.multichain.org/#/router

Here's the HMT contract on OKC:
https://www.oklink.com/okc/address/0x1351ab0323b1127e1cfb8406adf87ca88a5a4ff6

**Comment by Vonder85** [comment]
Hello, i begin to work on this issue.

**Comment by Vonder85** [comment]
I deployed all the contracts on the OKC TestNet:
here are adresses:

HMToken Address:  0x1DC0bcbF8eEcC1F3dEb9b33c0173E2e6ebB61419
Staking Proxy Address:  0x99a4256F0c4612851E8F20F06Fe9f4C550625b4D
Staking Implementation Address:  0x8739B52f44b41d97323ea962491bCd9fBa51e1ea
Escrow Factory Proxy Address:  0xC66968bD8C2C4319013ddBb2E6D1579EE35b3d1E
Escrow Factory Implementation Address:  0x18Aa40969F751D2C8Beacb27a5723EDf1d015A31
KVStore Address:  0xB56Cb67d37BcDb5Ec78B1FA63B83f789C1537699
Reward Pool Proxy Address:  0x0A85Bd0DeFd0a292B03F6275E5393F821f0eE3CD
Reward Pool Implementation Address:  0x006475518716e0EAC2301C3738F66881C0e8BaDC
Reputation Proxy Address:  0xe3Fe3150316DcfE7412e247E11124d09Bc8710C8
Reputation Implementation Address:  0x85cBCEFb028d806D15Fa79651BE7C634B24262e0

My next step step is to launch Fortune with these contracts

**Comment by foufrix** [comment]
Awesome keep us updated, and feel free to reach if you encounter any issue.

**Comment by Vonder85** [comment]
I deployed contracts on testNet:
HMT Address: 0x63170900372cCc15E21A48F037A774D92B93B6Be
Staking Proxy Address:  0x20ef0Ddd8F4712b37A518Fe671007d46B0C2511D
Staking Implementation Address:  0x19FA1D34008eB3375Ec83C54800bc068c85B10d6
Escrow Factory Proxy Address:  0x4FD64891D4F1a8c3a56011cDB6a7c41a69d253c5
Escrow Factory Implementation Address:  0x67870F28E699DF149b4B1D21AcE2D3613A01d1d4
KVStore Address:  0x3775efc35Cd8CcDAC9C4A41542703112F82cB6e0
Reward Pool Proxy Address:  0x99d4429862f8BCbC45Cc92A7f5f6d1AcB65CFB6E
Reward Pool Implementation Address:  0xE8929Ba05D149c172B2663775fc4E94dA0b3d27A

I fund with Fortune Launcher and submitted with the Exchange.
I did a video, tell me if there is something wrong please.

https://streamable.com/guubxp

The link for transactions in OKC Test Explorer:
https://www.oklink.com/fr/oktc-test/address/0xc2E20D0fBa93Cb57C2b1F98E194667fEe02b3d55

**Comment by foufrix** [comment]
Hi @Vonder85 , please check that you updated the address in this file to address you own : https://github.com/humanprotocol/human-protocol/blob/main/packages/examples/fortune/manifest.json

And be sure to have okx on it, and that all servers are running when you do your testing (recording, reputation, exchange oracale)
There should be another file where to put the private key of those addresses as well

When we search for the private key in readme : https://github.com/search?q=repo%3Ahumanprotocol%2Fhuman-protocol%20ac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80&type=code 
I'm not sure wich one should be change so that fortune understand what correct account to use. 

Be sure ton testnet/mainnet to not use those ones are they will be directly siphoned because private key are on github

**Comment by foufrix** [comment]
@Vonder85 each project has their own env file to modify:
- job launcher
- exchange oracle
- recording oracle
- reputation oracle

**Comment by foufrix** [comment]
Contract deployed on testnet, having some trouble with verify with OKCT at the moment but will find the issue

| 🟠 OKTC(Testnet) | Contract      | Address                                    | Proxy                                      |
| ---------------- | ------------- | ------------------------------------------ | ------------------------------------------ |
| 2023/05/26       | HMToken       | 0x854EC65E9e5e973C458FC2c92F6E0CbD403f5b95 | N/A                                        |
| 2023/05/26       | EscrowFactory | 0x56C2ba540726ED4f46E7a134b6b9Ee9C867FcF92 | 0x707fb5A5d36BC15275Af3f73262bf9a1D8C470EB |
| 2023/05/26       | Staking       | 0x9890473B0b93E24d6D1a8Dfb739D577C6f25FFd3 | 0xD8c35adC3b386d092846a93015220b7Fe8efD938 |
| 2023/05/26       | RewardPool    | 0x5517fE916Fe9F8dB15B0DDc76ebDf0BdDCd4ed18 | 0x2bfA592DBDaF434DDcbb893B1916120d181DAD18 |
| 2023/05/26       | EthKVStore    | 0xE3D74BBFa45B4bCa69FF28891fBE392f4B4d4e4d | N/A                                        |
| 2023/05/26       | Reputation    | N/A                                        | N/A                                        |

**Comment by foufrix** [comment]
Hi,

[Video of fortune working on testnet](https://www.loom.com/share/c35273c6cbc84dad922e4313a592d95d)

- [Escrow creation tx](https://www.oklink.com/fr/oktc-test/tx/0x6B80D540D1577C3FA39AFA25697E89CFB0A9EE31D3A33EA6E9573E57F859ECEE?tab=1)
- [Escrow Funding Tx](https://www.oklink.com/fr/oktc-test/tx/0x542f579b1b0980338671996f330ed9e0068ae38f5c33d20fc1dbf9e583745f10)
- [Escrow Setup](https://www.oklink.com/fr/oktc-test/tx/0x3E8DAFF5D1E2A10DAE7A7109B0A079BAEA3FB38D0DE9E8AFCFE76E22CFA62234)
- [Escrow Payout](https://www.oklink.com/fr/oktc-test/tx/0xdc2d53055e78d50b5452bb787f624f1980fa49ca6ebbcd50ad169e49acb95faf)

Once validated we can move on to mainnet deployment

**Comment by ansaqib** [comment]
It's assigned to you @foufrix

**Comment by portuu3** [comment]
> Hi,
> 
> [Video of fortune working on testnet](https://www.loom.com/share/c35273c6cbc84dad922e4313a592d95d)
> 
> * [Escrow creation tx](https://www.oklink.com/fr/oktc-test/tx/0x6B80D540D1577C3FA39AFA25697E89CFB0A9EE31D3A33EA6E9573E57F859ECEE?tab=1)
> * [Escrow Funding Tx](https://www.oklink.com/fr/oktc-test/tx/0x542f579b1b0980338671996f330ed9e0068ae38f5c33d20fc1dbf9e583745f10)
> * [Escrow Setup](https://www.oklink.com/fr/oktc-test/tx/0x3E8DAFF5D1E2A10DAE7A7109B0A079BAEA3FB38D0DE9E8AFCFE76E22CFA62234)
> * [Escrow Payout](https://www.oklink.com/fr/oktc-test/tx/0xdc2d53055e78d50b5452bb787f624f1980fa49ca6ebbcd50ad169e49acb95faf)
> 
> Once validated we can move on to mainnet deployment
> 
> Start of the [PR](https://github.com/humanprotocol/human-protocol/pull/641) not yet finished as we need prod deployment address as well

So far looks good to me.
Just please start marking done steps in checklist (added 2 last)

**Comment by portuu3** [comment]
Will be done in #1537

**Comment by foufrix** [comment]
I did this one already, was waiting for HMT to be deployed on OKT as there is no bridge currently

**Comment by portuu3** [comment]
@foufrix please can you summarize what has been done so far? 
who owns the keys used to deploy the contracts?

**Comment by foufrix** [comment]
I'm the owner of the key of the deployed contract.
I deployed everything on OKX's testnet, but they are now moving to another chain called X1 that supports zkEVM.

I was waiting for HMT to be deployed there. Will redeploy everything on X1 testnet (mainnet still not available)

Discussion with Worhmole are done to deploy HMT on mainnet once X1 is officially launched

**Comment by portuu3** [comment]
> @portuu3 I'm the owner of the key of the deployed contract. I deployed everything on OKX's testnet, but they are now moving to another chain called X1 that supports zkEVM.
> 
> I was waiting for HMT to be deployed there. Will redeploy everything on X1 testnet (mainnet still not available)
> 
> Discussion with Worhmole are done to deploy HMT on mainnet once X1 is officially launched

I can deploy HMT on testnet. Can you please provide RPC details?
